### PR TITLE
[AIR] Fix bug where `TensorflowPredictor.predict` creates extra axis

### LINKED ISSUE #25124: [AIR] `TensorflowPredictor` fails on multidimensional inputs
### What happened + What you expected to happen

I tried to classify an image, but my code errored.

```
❯ python /tmp/module.py
Metal device set to: Apple M1 Pro

systemMemory: 32.00 GB
maxCacheSize: 10.67 GB

2022-05-25 16:11:06.841758: I tensorflow/core/common_runtime/pluggable_device/pluggable_device_factory.cc:305] Could not identify NUMA node of platform GPU ID 0, defaulting to 0. Your kernel may not have been built with NUMA support.
2022-05-25 16:11:06.841894: I tensorflow/core/common_runtime/pluggable_device/pluggable_device_factory.cc:271] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 0 MB memory) -> physical PluggableDevice (device: 0, name: METAL, pci bus id: <undefined>)
Traceback (most recent call last):
  File "/tmp/module.py", line 32, in <module>
    predictor.predict(batch)
  File "/Users/bveeramani/GitHub/ray/python/ray/ml/predictors/integrations/tensorflow/tensorflow_predictor.py", line 150, in predict
    prediction = list(model(tensor).numpy())
  File "/Users/bveeramani/GitHub/ray/.venv/lib/python3.9/site-packages/keras/utils/traceback_utils.py", line 67, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "/Users/bveeramani/GitHub/ray/.venv/lib/python3.9/site-packages/keras/engine/input_spec.py", line 264, in assert_input_compatibility
    raise ValueError(f'Input {input_index} of layer "{layer_name}" is '
ValueError: Input 0 of layer "sequential" is incompatible with the layer: expected shape=(None, 32, 32, 3), found shape=(4, 1, 32, 32, 3)
```

### Versions / Dependencies

**Ray**: ac620aeec0c0f68c92328ace0b2a5835f5b14b26
**Python**: 3.8.12
**OS**: MacOS

### Reproduction script

```
from ray.ml.predictors.integrations.tensorflow import TensorflowPredictor
from tensorflow.keras import models, layers
import numpy as np

batch_size = 4
height = 32
width = 32
num_channels = 3
num_classes = 10


def build_model():
    model = models.Sequential()
    model.add(
        layers.Conv2D(
            6, (5, 5), activation="relu", input_shape=(height, width, num_channels)
        )
    )
    model.add(layers.MaxPooling2D((2, 2)))
    model.add(layers.Conv2D(16, (5, 5), activation="relu"))
    model.add(layers.MaxPooling2D((2, 2)))
    model.add(layers.Flatten())
    model.add(layers.Dense(120, activation="relu"))
    model.add(layers.Dense(84, activation="relu"))
    model.add(layers.Dense(num_classes))
    return model


predictor = TensorflowPredictor(model_definition=build_model)

batch = np.zeros(shape=(batch_size, height, width, num_channels))
predictor.predict(batch)
```

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by bveeramani**
**Important**: This PR is stacked on:
- [x] #25208 
 

## Why are these changes needed?

If you pass a multidimensional array as input to `TensorflowPredictor.predict`, your code will error because the `predict` method adds an extra axis to the input array. For more information about the bug, see #25124.

## Related issue number

Fixes #25124 

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/ml/predictors/integrations/tensorflow/tensorflow_predictor.py [hunk 1]
```diff
@@ -135,8 +132,6 @@ def build_model(self):
             if feature_columns:
                 data = data[feature_columns]
             data = data.values
-        else:
-            data = data[:, feature_columns]
```
Comment (bveeramani) [inline python/ray/ml/predictors/integrations/tensorflow/tensorflow_predictor.py:139]: This is the line that adds an extra axis.

File: python/ray/ml/tests/test_tensorflow_predictor.py [hunk 2]
```diff
@@ -43,7 +43,7 @@ def test_init():
     assert checkpoint_predictor.preprocessor == predictor.preprocessor
```
Comment (bveeramani) [inline python/ray/ml/tests/test_tensorflow_predictor.py:43]: Out-of-scope for this PR, but we should add tests for dataframes. Related to #25193.

File: python/ray/ml/tests/test_tensorflow_predictor.py [hunk 3]
```diff
@@ -53,25 +52,11 @@ def test_predict():
...
-    assert hasattr(predictor.preprocessor, "_batch_transformed")
-
-
-def test_predict_feature_columns():
-    preprocessor = DummyPreprocessor()
-    predictor = TensorflowPredictor(
-        model_definition=build_model, preprocessor=preprocessor, model_weights=weights
-    )
-
-    data_batch = np.array([[1, 4], [2, 5], [3, 6]])
```
[the remark this answers is not in the document]
Comment (bveeramani) [reply python/ray/ml/tests/test_tensorflow_predictor.py:66]: Fixed

**Review by amogkam — APPROVED** [review]
LGTM, thanks!

**Pull request merged**